guides/source/routing.md

```diff
@@ -738,7 +778,7 @@ You can use the `:constraints` option to enforce a format for a dynamic segment:
 get 'photos/:id', to: 'photos#show', constraints: { id: /[A-Z]\d{5}/ }
 ```
 
-This route would match paths such as `/photos/A12345`, but not `/photos/893`. You can more succinctly express the same route this way:
+The above route definition requires `id` to be 5 alphanumeric characters long. Therefore, this route would match paths such as `/photos/A12345`, but not `/photos/893`. You can more succinctly express the same route this way:
```

Comment:
The guide states that `:contraints` takes a regular expression, but I wonder if that's 100% correct. I think it accepts any object that responds to `#matches?` or `#call`

Reply:
Yes. We cover that in the "Advance Constraints" section below. It can take custom class that implements `matches?`, lambda, etc.

I added a note to this section.